packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
@mjackson I was experimenting with removing the `listen` calls (and adding one to `<Router>`), and it was quite pleasant. Added a `react-router-listen` package for people who use `shouldComponentUpdate`.

It really does make things much nicer. Only the `<Router>` and `listen`/`<Listener>` have to listen for navigation, everything else receives the new location through the normal render lifecycle. Switch animations work without having to add a location prop.

https://github.com/pshrmn/react-router/tree/im-not-listening

Unless you see yourself removing the automatic support for getting around `sCU`, I won't submit a PR, but it does show how much simpler things could be.